api/include/opentelemetry/config.h

```diff
@@ -9,6 +9,6 @@
 #  define OPENTELEMETRY_HAS_INCLUDE(x) __has_include(x)
 #endif
 
-#if !defined(__GLIBCXX__) || OPENTELEMETRY_HAS_INCLUDE(<codecvt>)  // >= libstdc++-5
+#if defined(OPENTELEMETRY_STL_VERSION) && OPENTELEMETRY_STL_VERSION > 2011
```

Comment:
The test should probably be `>= 2011`.

Now, given that `OPENTELEMETRY_TRIVIALITY_TYPE_TRAITS` is used in exactly one place, and given that it relie on the C++11 std headers anyway, this can be simplified further:

* please remove the code that sets `OPENTELEMETRY_TRIVIALITY_TYPE_TRAITS` in this config.h file
* please change the only place where it is used with `ifdef OPENTELEMETRY_HAVE_STD_TYPE_TRAITS` instead.

Reply:
Did as suggested